server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -232,7 +232,7 @@ void deprecated(final Set<ThreadContext> threadContexts, final String message, f
             while (iterator.hasNext()) {
                 try {
                     final ThreadContext next = iterator.next();
-                    next.addResponseHeader("Warning", warningHeaderValue, DeprecationLogger::extractWarningValueFromWarningHeader);
```

Comment:
I wonder whether we need `DeprecationLogger::extractWarningValueFromWarningHeader` at all now? It seems to be used in some tests and an assert in `DeprecationLogger` but I think we can get rid of it there as well?

Then we can also get rid of `ThreadContext#addResponseHeader(final String key, final String value, final Function<String, String> uniqueValue)` and just have `#addResponseHeader(final String key, final String value)`?

In order to keep this PR small this could be done in a follow-up though as well.

Reply:
Perfect. Then I'll approve now.